lua/entities/starfall_cnextbot.lua

```diff
@@ -0,0 +1,156 @@
+		
+		if ( options.draw ) then path:Draw() end
+		-- If we're stuck then call the HandleStuck function and abandon
+		if ( self.loco:IsStuck() ) then
+			self:HandleStuck()
+			return "stuck"
+		end
+
+		coroutine.yield()
+
+	end
+
+	return "ok"
+end
+
+function ENT:RunBehaviour()
+	
+	while true do
+		if self.playSeq then
+			self:PlaySequenceAndWait( self.playSeq )
+			self.playSeq = nil
+		elseif self.goTo then
+			self.loco:SetDesiredSpeed(self.MoveSpeed)
+			self:StartActivity(self.RUNACT)
+			self:ChasePos()
+			self:StartActivity(self.IDLEACT)
+			self.goTo = nil
+		end
+		
+		coroutine.wait( 1 )
+		
+		coroutine.yield()
+	end
+	
+end
+
+-- TO DO: Maybe look into caching the table counts as I don't know how expensive the counting function is
+function ENT:OnInjured(dmginfo)
+	if self.InjuredCallbacks:isEmpty() then return end
+	local inst = self.chip.instance
+	self.InjuredCallbacks:run(inst,
+		dmginfo:GetDamage(),
+		inst.Types.Entity.Wrap(dmginfo:GetAttacker()),
+		inst.Types.Entity.Wrap(dmginfo:GetInflictor()),
+		inst.Types.Vector.Wrap(dmginfo:GetDamagePosition()), 
+		inst.Types.Vector.Wrap(dmginfo:GetDamageForce()),
+		dmginfo:GetDamageType())
+end
+
+function ENT:OnKilled(dmginfo)
+	if self.DeathCallbacks:isEmpty() then return end
+	local inst = self.chip.instance
+	self.DeathCallbacks:run(inst,
+		dmginfo:GetDamage(),
+		inst.Types.Entity.Wrap(dmginfo:GetAttacker()),
+		inst.Types.Entity.Wrap(dmginfo:GetInflictor()),
+		inst.Types.Vector.Wrap(dmginfo:GetDamagePosition()),
+		inst.Types.Vector.Wrap(dmginfo:GetDamageForce()),
+		dmginfo:GetDamageType())
+	if self.RagdollOnDeath then self:BecomeRagdoll(dmginfo) end
+end
+
+function ENT:OnLandOnGround(groundent)
+	if self.LandCallbacks:isEmpty() then return end
+	local inst = self.chip.instance
+	self.LandCallbacks:run(inst, inst.Types.Entity.Wrap(groundent))
+end
+
+function ENT:OnLeaveGround(groundent)
+	if self.JumpCallbacks:isEmpty() then return end
+	local inst = self.chip.instance
+	self.JumpCallbacks:run(inst, self.chip.instance.Types.Entity.Wrap(groundent))
+end
+
+function ENT:OnIgnite()
+	if self.IgniteCallbacks:isEmpty() then return end
+	self.IgniteCallbacks:run(inst)
+end
+
+function ENT:OnNavAreaChanged(old, new)
+	if self.NavChangeCallbacks:isEmpty() then return end
+	local inst = self.chip.instance
+	self.NavChangeCallbacks:run(inst,
+		inst.Types.NavArea.Wrap(old),
+		inst.Types.NavArea.Wrap(new))
+end
+
+function ENT:OnContact(colent)
+	if self.ContactCallbacks:isEmpty() then return end
+	local inst = self.chip.instance
+	self.ContactCallbacks:run(inst, inst.Types.Entity.Wrap(colent))
+end
+
+function ENT:BodyUpdate()
+	self:BodyMoveXY()
+	
+	local velForw = self:EyeAngles():Forward():Dot(self.loco:GetVelocity())
+	local velRight = self:EyeAngles():Right():Dot(self.loco:GetVelocity())
+	local range = math.atan2(-velRight, velForw) / math.pi
+	local remappedRange = math.Remap(range, -1, 1, -180, 180)
+	self:SetPoseParameter("move_yaw", remappedRange )
```

Comment:
This can be simplified
```lua
local localVel = self:WorldToLocal(self.loco:GetVelocity() + self:GetPos())
self:SetPoseParameter("move_yaw", math.deg(math.atan2(localVel.x, localVel.y)))
```